### PR TITLE
Flex 1661 port password component to react

### PR DESCRIPTION
**Pull request by akate1**
image

image

### CODE REVIEW

File: src/forms/password/Password.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                    {label && Password.renderLabel(label)}
+                    <div className={utils.classNames(classNames)} style={{width: "100%"}}>
+                        <div className="clr-input-wrapper">
+                            <div className="clr-input-group clr-row" style={{...style, paddingRight: "1rem"}}>
+                                <input
+                                    maxLength={maxPasswordLength}
+                                    minLength={minPasswordLength}
+                                    name={name}
+                                    defaultValue={defaultValue}
+                                    value={value}
```
Comment (psivakumaran) [inline src/forms/password/Password.tsx]: When we enter the value in the password component, and if I do not have a onChange which sets the value prop explicitly, how will I be able to get the password value? I think value should be part of the password state

File: src/forms/password/Password.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                <div className={utils.classNames(["clr-form-control", label && "clr-row"])}>
+                    {label && Password.renderLabel(label)}
+                    <div className={utils.classNames(classNames)} style={{width: "100%"}}>
+                        <div className="clr-input-wrapper">
+                            <div className="clr-input-group clr-row" style={{...style, paddingRight: "1rem"}}>
+                                <input
+                                    maxLength={maxPasswordLength}
+                                    minLength={minPasswordLength}
+                                    name={name}
+                                    defaultValue={defaultValue}
```
Comment (psivakumaran) [inline src/forms/password/Password.tsx]: Generally we should not have a default value for password

File: src/forms/password/Password.tsx (hunk 1 above)
Comment (akate1) [reply src/forms/password/Password.tsx]: done

File: src/forms/password/Password.tsx (hunk 2 above)
Comment (akate1) [reply src/forms/password/Password.tsx]: keeping this for now. There may be scenario where we get password from back end and user need to change it.

Comment (psivakumaran) [reply src/forms/password/Password.tsx]: Agreed

**Pull request merged**